### DISCUSSION TITLE
WiFi Support

### DISCUSSION THREAD

**Discussion by AdamKearn** [ipxe/ipxe#851] [Q&A]
Hey guys,

I've been spending quite a bit of time doing some research over the last week or so.
I'm trying to get WiFi working directly in iPXE.

A lot of the forum posts regarding this are over 10 years old...  I've found a few that have mentioned of them getting it working but sadly not proving the details on replicating it...   Currently using the master branch I've had zero success getting WiFi working in my environment.    (using an embedded script to set SSID/KEY then autobooting...)

I just want to get an understanding of what the current plan is for getting WiFi added.  (not USB tethering)

As more and more of these modern computers are getting released without having a NIC/LAN port to keep the laptop thin.
Which can be quite problematic when deploying a large number of computers.

I'm assuming the whole code for WPA/WPA2 is needing to be rewritten.

Regards,
Adam.

**Comment by NiKiZe** [comment]
There is only a very few limited drivers for actual WiFi devices. You need one of those NICs to get it running.

Does the NIC show up in iPXE if you run the ifstat command in iPXE shell? What is the pciid of devices you tried?

One major issue with modern hardware is often the requirement for binary blobs to get them working.

However some devices now come with WiFi support in firmware. If such you can use SNP/nii driver.

You mention that you don't want to use USB tethering. Unfortunately that is probably the best way. However you might also want to consider USB NICs with cable.
A good machine without ethernet jack's do have support for some USB NICs in firmware.


Possible/Likely duplicate of what is discussed i. The thread with your comment https://github.com/ipxe/ipxe/discussions/354#discussioncomment-4668071

**Reply by AdamKearn** [reply]
Sorry for the delay in responding have been a bit busy the last few days.

> Does the NIC show up in iPXE if you run the ifstat command in iPXE shell? What is the pciid of devices you tried?

No, that was one of the issues I was facing.   when running `ifstat` it ended up showing zero output.
I wasn't sure if this was just because I was using the `EFI` edition so I downloaded the basic `BIOS` version and had the same outcome.

To be honest most of the commands other than `help` and `chain ...` don't end up outputting anything to the console.
This is when pressing `CTRL + B` when booting into iPXE.

> However you might also want to consider USB NICs with cable.

Yeah that's what I am currently using.   I was able to build iPXE with the driver that our USB Adapters use.     (`ax88140 / ASIX` is what I needed) 
Using the dongle works fine its just a bit of a pain when trying to do 30+ https://www.amazon.co.uk/gp/product/B0095EFXMC

The laptops I am trying to image have just a generic intel wifi chip   (don't have the model number of the chip to hand at the moment)

---

My main goal would be to get the WiFi completely working.  Regarding what you mentioned with hardware needing "binary blobs".
Are you unable to support them natively due to licence issues?

If we could even work together to build a generic guide to help people in the same situation as I am.
So they can too also build iPXE from source and add their required drivers that would be great.

**Reply by bam80** [reply]
> However some devices now come with WiFi support in firmware. If such you can use SNP/nii driver.

Could you expand on that, which devices? Is it really practical?
